src/mixins/filing-template-mixin.ts

```diff
@@ -483,74 +483,71 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
    * @param entitySnapshot the latest entity snapshot
    */
   parseCorrectionFiling (filing: CorrectionFilingIF, entitySnapshot: EntitySnapshotIF = null): void {
-    // make a copy so snapshot objects are different from objects below
-    entitySnapshot = cloneDeep(entitySnapshot)
-
     // store Entity Snapshot
     this.setEntitySnapshot(entitySnapshot)
 
     // store Correction Information
-    this.setCorrectionInformation(filing.correction)
+    this.setCorrectionInformation(cloneDeep(filing.correction))
```

Comment:
For objects that have sub-objects, make a copy using cloneDeep().

Reply:
So if you don't use cloneDeep does this setter only set the top level object

Reply:
Not exactly.

If you don't use cloneDeep then this setter sets its object reference to this object reference. In other words, `state.stateModel.correctionInformation` will point to the same object as `filing.correction`, so if you change something in one then the other one sees the same thing.

CloneDeep create a new object so that each variable points to its own object.

Reply:
To amplify my first comment above, if I just did `this.setCorrectionInformation({ ... filing.correction })` then that would do a _shallow_ copy, so any sub-objects would still have the same reference as the original object. That can lead to really hard-to-find bugs.